src/renderer/rendered_window.rs

```diff
@@ -721,6 +705,61 @@ impl RenderedWindow {
         self.scroll_delta = 0;
     }
 
+    fn iter_lines<'a>(&'a self) -> Box<dyn Iterator<Item = &'a Rc<RefCell<Line>>> + 'a> {
```

Comment:
If this is split into three different iterators, then I think these iterators could be used as the base for, the `split_lines` (or the refactored new functions) as well.